examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -129,47 +131,92 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
+        // Can be removed once all other platforms (ESP, etc.)
+        // start using the ota-candidates-file method.
+        if (strlen(mOTAFilePath)) // If OTA file is directly provided
         {
-            newSoftwareVersion       = candidate.softwareVersion;
-            newSoftwareVersionString = candidate.softwareVersionString;
-            otaFilePath              = candidate.otaURL;
+            newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
+                                                                  // available (if the user provides a file).
+            newSoftwareVersionString = "Example-Image-V0.1";
+            otaFilePath              = mOTAFilePath;
+            queryStatus              = OTAQueryStatus::kUpdateAvailable;
+        }
+        else if (!mCandidates.empty()) // If list of OTA candidates is supplied instead
+        {
+            if (SelectOTACandidate(commandData.vendorId, commandData.productId, commandData.softwareVersion, candidate))
+            {
+                newSoftwareVersion       = candidate.softwareVersion;
+                newSoftwareVersionString = candidate.softwareVersionString;
+                otaFilePath              = candidate.otaURL;
+                queryStatus              = OTAQueryStatus::kUpdateAvailable;
+            }
         }
-    }
 
-    GenerateUpdateToken(updateToken, kUpdateTokenLen);
-    GetUpdateTokenString(ByteSpan(updateToken), strBuf, kUpdateTokenStrLen);
-    ChipLogDetail(SoftwareUpdate, "generated updateToken: %s", strBuf);
+        if (queryStatus == OTAQueryStatus::kUpdateAvailable && mUserConsentDelegate != nullptr)
+        {
+            UserConsentState state = mUserConsentDelegate->GetUserConsentState(commandObj->SourceNodeId(), commandPath.mEndpointId,
+                                                                               commandData.softwareVersion, newSoftwareVersion);
+            switch (state)
+            {
+            case UserConsentState::kGranted:
+                queryStatus = OTAQueryStatus::kUpdateAvailable;
+                break;
+
+            case chip::ota::UserConsentState::kObtaining:
+                queryStatus          = OTAQueryStatus::kBusy;
+                delayedActionTimeSec = (delayedActionTimeSec < 120) ? 120 : delayedActionTimeSec;
+                break;
+
+            case chip::ota::UserConsentState::kDenied:
+                queryStatus          = OTAQueryStatus::kNotAvailable;
+                delayedActionTimeSec = (delayedActionTimeSec < 120) ? 120 : delayedActionTimeSec;
+                break;
+            }
+        }
+        break;
+
+    case kRespondWithUpdateAvailable:
+        queryStatus = OTAQueryStatus::kUpdateAvailable;
+        break;
+
+    case kRespondWithBusy:
+        queryStatus          = OTAQueryStatus::kBusy;
+        delayedActionTimeSec = (delayedActionTimeSec < 120) ? 120 : delayedActionTimeSec;
```

Comment:
The entire ternary business could be entirely removed from every `case` and moved past the switch statement to a simple:

`delayedActionTimeSec = std::max(kMininumDelayedActionTime, delayedActionTimeSec);`